core/src/main/java/com/novoda/noplayer/internal/mediaplayer/AndroidMediaPlayerImpl.java

```diff
@@ -333,6 +333,7 @@ public void release() {
     }
 
     private void reset() {
+        handler.removeCallbacksAndMessages(null);
```

Comment:
@Mecharyry Isn't this risky? If more runnables are posted to that handler in the future you'll cancel all of them here without explicitly saying so. Maybe that's desirable but maybe is not.
There's also `handler.removeCallbacks(runnable)` where you need to specify the runnable you want to kill

Reply:
👍 would be safer to keep a reference to the runnable and cancel that instead _orrrrrr_ create a new handler which is only used for this workaround, then we should be good to nuke it.

Reply:
I did consider this. Ultimately doing `handler.removeCallbacks(runnable)` would require me to keep track of all `runnables` that this class can produce. Really this class should never have more than one runnable in play at any one time, the delay is short and there are a few conditionals that would need to be triggered to make this happen. 

Perhaps the biggest reason for not using `handler.removeCallbacks(runnable)` is the purpose of the `reset`. `reset` is used to restore the `MediaPlayerImpl` to it's default state, that is, the state it contains immediately after creation. In creation the `MediaPlayerImpl` does not contain a `handler` with `runnables` and therefore it shouldn't after a `reset` is called. 

Thanks for mentioning it ❤️  it's nice to explain this to put it in my head more 😄

Reply:
I'll opt to rename and potentially wrap this handler to make this more explicit, would prefer over creating new handlers or keeping track of runnables.